### ISSUE TITLE
[Bug]: IEventSource3.IncludeTaskInputs() doesnt actually include Task Input events

### ISSUE THREAD

**Issue by baronfel** [dotnet/msbuild#9804]
### Issue Description

In the [`use-sourceroot-for-relative-paths`](https://github.com/dotnet/msbuild/tree/use-sourceroot-for-relative-paths) branch I'm working on extending Terminal Logger to read the SourceRoot items in the build to discover the current git repo, if any. This is done  by

a) [opting in to Task Input events](https://github.com/dotnet/msbuild/blob/use-sourceroot-for-relative-paths/src/MSBuild/TerminalLogger/TerminalLogger.cs#L254-L257)
b) [reacting to the TaskParameterEventArgs events](https://github.com/dotnet/msbuild/blob/use-sourceroot-for-relative-paths/src/MSBuild/TerminalLogger/TerminalLogger.cs#L702-L712)

However, the opt-in from `IEventSource3.IncludeTaskInputs()` isn't causing the events to be fired.

### Steps to Reproduce

* Clone the [`use-sourceroot-for-relative-paths`](https://github.com/dotnet/msbuild/tree/use-sourceroot-for-relative-paths) branch
* Build msbuild with `./build.cmd`
* Copy MSbuild to an 8.0.2xx .NET SDK installation with the enlistment scripts
* Set MSBUILDDEBUGONSTART to 2
* build a project that's in a git repo
* attach a debugger to line 705 in the TerminalLogger.cs file
* see the breakpoint is never hit

### Expected Behavior

TaskParameterEventArgs events should be sent to the logger

### Actual Behavior

No TaskParameterEventArgs events are fired

### Analysis

Per @ladipro, LoggingService.IncludeTaskInputs is not plumbed to interact with BuildParameters.LogTaskInputs.

**Comment by AR-May** [comment]
@ladipro could you please comment if we need to change this behavior?

**Comment by ladipro** [comment]
@AR-May I still think it would be good to fix this. It looks like the original change in #3253 made us log task inputs if at least one logger requested it, but it was done only for VS. Basically, we need the equivalent of this code to exist on the code path where we do CLI builds:

https://github.com/dotnet/msbuild/blob/8526e850a54170457000221aafb36236a7077062/src/Build/Instance/ProjectInstance.cs#L2488-L2494

Though, if it's not blocking for @baronfel, I guess it can be parked in backlog

**Comment by baronfel** [comment]
Yes, this is a correctness/completion fix - if I had this I would be able to more accurately implement my change.. right now I'm having to use a heuristic, but I'd really rather be able to get the outputs from a specific Task.
